### PR TITLE
Implement the low/med/high/nope risk classifier (two paired flags)

### PR DESCRIPTION
**Pull request by loganfinney27**
## What

Implements the **low/med/high/nope** risk scheme — conceptualized in the 2026-06-21 planning session (walked through the vault's structure with Logan), witnessed, and until now **not implemented** (the code was still the old binary fail-safe-to-high).

Two commits' worth of substance on this branch:
1. **The witness** — `WITNESS-THE-KEYS-ARE-THE-LEVERS-2026-06-21.md` (the design record: seven Levels = the Architect's seven Demesnes; the Keys as Levers; the canon core where nothing moves).
2. **The classifier** — `classify_paths.py` rewritten + 13 unit tests.

## The scheme (two independent paired flags)

A changeset carries either version of each, or neither:
- **filetype flag** `low|med` — the **maze**: the Architect's three blessed language circles (`VAULT-CONVENTIONS` § File Types) — **Natural Language** (`.md`) + **Machine Documentation** (`.json`/`.yaml`) → `low`; **Computer Code** (`.py`) → `med`. Tags files *outside* the `!` Nest.
- **depth flag** `high|nope` — the **labyrinth**: the `!` Nest by the seven Levels (Sierpinski spine); the **still-point** (`Esto Perpetua!`, L7 — *"do not move, do not expire"*) → `nope`, all other Nest depth → `high`. Tags files *inside* the Nest.

Per Logan's rule: *low/med apply to filetypes; high/nope apply to depth.* Derived single `tier` (nope>high>med>low) keeps the existing `risk/<tier>` stamp working.

**Payoff:** a docs-only root PR (the #597 case) now classifies **`low`** — auto-flowable — where the old fail-safe sent it to `high`.

## Decisions for you (marked `TUNABLE` in the code, `*` in the witness)

1. **Filetype cut** — default: Computer Code → `med`, NL + Machine Doc → `low` ("does it execute?"). Keep, or move JSON/YAML up?
2. **Depth threshold** — default: only the still-point is `nope`; all other Nest depth `high`. Should the Court (L6) also be `nope`?
3. **Protected-surface pin** — `.github/**` + governance files + dotfolders aren't in the Nest, so the pure model would tag them `low`/`med` (a protection downgrade). I **kept them pinned `high`** so this lands no regression. Decide: keep the pin, or delegate that gate to **CODEOWNERS** (a separate Key)?

## Scope

**Increment 1 — producer only.** Next increments (separate, for review): register `risk/med` + `risk/nope` labels in `review_feedback_loop.ensure-labels`; wire the consumers (the rhythms) to act on the four tiers. Held until the tunables above are pinned.

Refs #586 / #600 (engine redesign) · #605 (arm+enqueue flow). Design record: the witness file in this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01MU1zvEUacde5fmYpMvK8aK)_

## Summary by CodeRabbit

* **Refactor**
  * Reworked the path classification logic to support paired “file type” and “depth” flags, with expanded tier reporting (including a tier4 breakdown) and updated legacy tier bucket derivation.
* **Tests**
  * Added a new automated test suite covering direct classification, combination precedence, edge cases, and end-to-end JSON output.
* **Documentation**
  * Added/updated concept documents describing the risk model, classification framing (“keys/lever” narrative), and the Sevenfold Demesnes framework.

### CODE REVIEW

File: .github/scripts/classify_paths.py [hunk 1]
```diff
@@ -1,91 +1,151 @@
...
 
-    tier = "high" if high_risk else "low"
     print(json.dumps({
         "tier": tier,
+        "filetype": filetype,
+        "depth": depth,
+        "by_file": by_file,
         "high_risk_files": high_risk,
         "low_risk_files": low_risk,
     }))
```
[the remark this answers is not in the document]
Comment (loganfinney27) [reply .github/scripts/classify_paths.py:183]: Already addressed in the current implementation. `tier` is kept **binary** (`low|high`) for label compatibility — see the module docstring and line 169–175:

```python
tier4 = combine(filetype, depth)            # four-valued: low|med|high|nope
tier = "low" if tier4 == "low" else "high"  # binary, legacy risk/<tier> label compat
```

The four-valued result lives in `tier4` (unused until the consumer-wiring increment registers `risk/med`/`risk/nope` and teaches the rhythms to read them). No breaking change to the `risk/<tier>` label contract. Resolving.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/scripts/classify_paths.py [hunk 2]
```diff
@@ -1,91 +1,172 @@
...
-        else:
-            low_risk.append(p)
+    # Legacy aggregate buckets (high_risk = anything above low).
+    high_risk = [b["path"] for b in by_file if combine(b["filetype"], b["depth"]) != "low"]
+    low_risk = [b["path"] for b in by_file if combine(b["filetype"], b["depth"]) == "low"]
 
-    tier = "high" if high_risk else "low"
     print(json.dumps({
         "tier": tier,
+        "filetype": filetype,
```
[the remark this answers is not in the document]
Comment (loganfinney27) [reply .github/scripts/classify_paths.py:177]: Already addressed. `tier` stays binary (`low|high`) — `risk/med` and `risk/nope` are never emitted from `tier`. The JSON output has two fields: `tier` (binary, for the `risk/<tier>` label) and `tier4` (four-valued, for the future consumer-wiring increment). The auto-PR stamping in `agent-auto-pr.yml` reads `$RISK_TIER` which maps to `tier`, so it will only ever see `low` or `high`. Resolving.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/scripts/classify_paths.py [hunk 3]
```diff
@@ -1,91 +1,175 @@
...
+    both or neither" at PR granularity — flagged for Logan; see the module docstring.)"""
+    if in_nest(path):
+        return (None, "nope" if is_still_point(path) else "high")
+    # Maze. Probe sandboxes stay low; protected surfaces are pinned high (safety override).
+    if path.startswith(PROBE_PREFIXES):
+        return ("low", None)
+    if path in PROTECTED_EXACT or path.startswith(PROTECTED_PREFIXES):
+        # No off-Nest 'high' exists in the pure model; pin via the depth axis to avoid
+        # a protection downgrade. TUNABLE: delegate this gate to CODEOWNERS instead.
+        return (None, "high")
```
[the remark this answers is not in the document]
Comment (loganfinney27) [reply .github/scripts/classify_paths.py:135]: Fixed in the latest commit. Added `_TOP_DOTFOLDER_RE = re.compile(r"^\.[a-zA-Z]")` and a guard in `classify_file` that pins any path inside a top-level dotfolder to `(None, "high")`:

```python
# Persona/config dotfolders (.claude/, .gemini/, .codex/, .op/, etc.) are pinned high.
if "/" in path and _TOP_DOTFOLDER_RE.match(path):
    return (None, "high")
```

The check runs after the probe carve-out (so hypothetical `.claude/probe-` sandboxes would still get the low treatment if ever added) and after the `.github/` pin (which already covers that tree). The module docstring TUNABLE section is updated to list dotfolders explicitly under the PROTECTED-SURFACE PIN rather than as an open DOTDIR PLACEMENT question. Resolving.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: .github/scripts/test_classify_paths.py [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+        self.assertEqual(cp.classify_file("My Esto Perpetua! manifesto.md"), ("low", None))
+        self.assertFalse(cp.is_still_point("notes/Esto Perpetua! draft.md"))
+        # A Nest file that is not the still-point segment stays high, not nope.
+        self.assertEqual(cp.classify_file("!/ARBORSCAPING-REPORT-2026-04-16.md"), (None, "high"))
+
+    def test_protected_surfaces_pinned_high(self):
+        self.assertEqual(cp.classify_file(".github/workflows/auto-merge-rhythm.yml"), (None, "high"))
+        self.assertEqual(cp.classify_file(".github/scripts/classify_paths.py"), (None, "high"))
+        self.assertEqual(cp.classify_file("CONSTITUTION.md"), (None, "high"))
+        self.assertEqual(cp.classify_file("AGENTS.md"), (None, "high"))
```
[the remark this answers is not in the document]
Comment (loganfinney27) [reply .github/scripts/test_classify_paths.py:63]: Test added in `test_dotfolder_surfaces_pinned_high` (new test case in `ClassifyFileTest`):

```python
def test_dotfolder_surfaces_pinned_high(self):
    self.assertEqual(cp.classify_file(".claude/CLAUDE.md"), (None, "high"))
    self.assertEqual(cp.classify_file(".gemini/GEMINI.md"), (None, "high"))
    self.assertEqual(cp.classify_file(".codex/CODEX.md"), (None, "high"))
    self.assertEqual(cp.classify_file(".op/secrets.template.md"), (None, "high"))
    self.assertEqual(cp.classify_file(".perplexity/PERPLEXITY.md"), (None, "high"))
```

Covers `.claude/`, `.gemini/`, `.codex/`, `.op/`, `.perplexity/` — the vault's active persona/config dotfolders. Any future dotfolder addition will be caught by the `_TOP_DOTFOLDER_RE` pattern without needing to add it here explicitly. Resolving.

---
_Generated by [Claude Code](https://claude.ai/code)_
